docker/ubuntu/Dockerfile

```diff
@@ -11,31 +14,42 @@ RUN apt-get update && apt-get install -y \
     git \
     build-essential \
     pkg-config \
-    python3 \
-    python3-dev \
-    python3-pip \
-    && pip3 install --upgrade pip \
-    && pip3 install pyarrow
+    python3.10 \
+    python3.10-dev \
+    python3-pip
 
 # Install Rust.
 SHELL ["/bin/bash", "-c"]
 RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --no-modify-path --default-toolchain none -y
 ENV PATH /root/.cargo/bin/:$PATH
 
 # Build the project in release mode.
-COPY . .
-RUN cargo build --release
+RUN --mount=target=.,rw \
+    --mount=type=cache,target=/usr/local/cargo/registry \
+    make build \
+    CARGO_PROFILE=${CARGO_PROFILE} \
+    FEATURES=${FEATURES} \
+    TARGET_DIR=/out/target
 
 # Export the binary to the clean image.
 # TODO(zyy17): Maybe should use the more secure container image.
 FROM ubuntu:22.04 as base
 
+ARG CARGO_PROFILE
+
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get \
     -y install ca-certificates \
+    python3.10 \
+    python3.10-dev \
+    python3-pip \
     curl
 
+COPY ./docker/python/requirements.txt /etc/greptime/requirements.txt
+
+RUN python3 -m pip install -r /etc/greptime/requirements.txt
```

Comment:
I'm curious why do we need to install pip requirements in Ubuntu but not in CentOS?

cc @zyy17